### PR TITLE
Resolves #1432 blockcmd addition (-a)

### LINKED ISSUE #1432: ./blockcmd addition
Usage: /blockcmd <purge> <partialname> <-a> the -a bit will block the commands of all players currently ingame who are not super admins.

### PR DESCRIPTION
**Pull request by LegendIsAwesomes**
Resolves #1432 blockcmd addition (-a)

### CODE REVIEW

**Comment by LegendIsAwesomes** [comment]
@JeromSar this compiles, has been tested etc, already done it all.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_blockcmd.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import org.bukkit.entity.Player;
 
 @CommandPermissions(level = AdminLevel.SUPER, source = SourceType.BOTH)
-@CommandParameters(description = "Block all commands for a specific player.", usage = "/<command> <purge | <partialname>>", aliases = "blockcommands,blockcommand")
+@CommandParameters(description = "Block all commands for a specific player.", usage = "/<command> <purge | <partialname> <-a>>", aliases = "blockcommands,blockcommand")
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_blockcmd.java]: -a is an optional parameter. Additionally, it defined wrongly here.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_blockcmd.java [hunk 2]
```diff
@@ -35,6 +35,22 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
+            TFM_PlayerData playerdata;
+            
+            TFM_Util.adminAction(sender.getName(), "Blocking commands for all non-Superadmins.", true);
+            
+            for (Player player : server.getOnlinePlayers())
+            {
+                if (!TFM_AdminList.isSuperAdmin(player))
+                {
+                    playerdata = TFM_PlayerData.getPlayerData(player);
+                    playerdata.setMuted(true);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_blockcmd.java]: Yeah right.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_blockcmd.java [hunk 3]
```diff
@@ -35,6 +35,22 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
+        {
+            TFM_PlayerData playerdata;
+            
+            TFM_Util.adminAction(sender.getName(), "Blocking commands for all non-Superadmins.", true);
+            
+            for (Player player : server.getOnlinePlayers())
+            {
+                if (!TFM_AdminList.isSuperAdmin(player))
+                {
+                    playerdata = TFM_PlayerData.getPlayerData(player);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_blockcmd.java]: Single use variable declared in a different scope.

**Comment by JeromSar** [comment]
Another shameful example of an untested feature. :/ @LegendIsAwesomes

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_blockcmd.java [hunk 4]
```diff
@@ -35,6 +35,22 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
             playerMsg("Unblocked commands for " + counter + " players.");
             return true;
         }
+        
+        if (args[0].equalsIgnoreCase("-a"))
+        {
+            TFM_PlayerData playerdata;
+            
+            TFM_Util.adminAction(sender.getName(), "Blocking commands for all non-Superadmins.", true);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_blockcmd.java]: Superadmins needs to be lowercase here. Also, no trailing full stop.

**Comment by LegendIsAwesomes** [comment]
@JeromSar fixed it.

**Comment by LegendIsAwesomes** [comment]
@jeromsar?

**Comment by Wild1145** [comment]
Looks good to me.

Just one thing for the future, please try to use descriptive commit names / descriptions, it makes it easier when it is merged to find your commits again.

**Comment by LegendIsAwesomes** [comment]
Okay will do @Wild1145

**Comment by DUFCLiam** [comment]
@JeromSar final review?

**Comment by LegendIsAwesomes** [comment]
@JeromSar I just checked and tested this it compiles and works as I tested it on ImmaFreedom which uses TFM.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_blockcmd.java [hunk 5]
```diff
@@ -8,7 +8,7 @@
 import org.bukkit.entity.Player;
 
 @CommandPermissions(level = AdminLevel.SUPER, source = SourceType.BOTH)
-@CommandParameters(description = "Block all commands for a specific player.", usage = "/<command> <purge | <partialname>>", aliases = "blockcommands,blockcommand")
+@CommandParameters(description = "Block all commands for a specific player.", usage = "/<command> <purge | <partialname> [-a]>", aliases = "blockcommands,blockcommand")
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_blockcmd.java]: `purge` and `partialname` become optional now that `-a` is added. Please make this is a three-way OR clause.

**Comment by JeromSar** [comment]
Just needs a small edit, than it's good to go. :)

**Comment by LegendIsAwesomes** [comment]
@JeromSar

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_blockcmd.java [hunk 6]
```diff
@@ -8,7 +8,7 @@
 import org.bukkit.entity.Player;
 
 @CommandPermissions(level = AdminLevel.SUPER, source = SourceType.BOTH)
-@CommandParameters(description = "Block all commands for a specific player.", usage = "/<command> <purge | <partialname>>", aliases = "blockcommands,blockcommand")
+@CommandParameters(description = "Block all commands for a specific player.", usage = "/<command> [purge] | [partialname] [-a]>", aliases = "blockcommands,blockcommand")
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_blockcmd.java]: This is not exactly what I meant. This way, it is suggested `purge`, `partialname` and `-a` can be used independently of one another, but all three are optional. This is not quite right. The correct line should be.

`"/<command> <-a | purge | partialname>"

**Comment by LegendIsAwesomes** [comment]
Ahh ye my mistake; I sometimes forget which ones mean optional etc.

**Comment by LegendIsAwesomes** [comment]
Like that @JeromSar

**Comment by JeromSar** [comment]
@LegendIsAwesomes That's perfect. Thanks for contributing! This PR will be merged soon (TM).

**Comment by LegendIsAwesomes** [comment]
Yay :P 6 PR's ready to merge.

**Review by Wild1145 — APPROVED** [review]
Should be ready to merge.

**Comment by JeromSar** [comment]
This feature was already added added in TFM Electrum.

**Pull request closed without merging**